### PR TITLE
Add TypeScript version of extensions_putting_together.py

### LINKED ISSUE #6596: Add TypeScript version of extensions_putting_together.py
For reference https://groups.google.com/a/continuum.io/forum/#!topic/bokeh/BEuBfJIWjhA.

### PR DESCRIPTION
**Pull request by mattpap**
fixes #6596

### CODE REVIEW

**Comment by mattpap** [comment]
@bryevdv, can you finish this (incorporate this into `user_guide/extensions.rst`)? I'm pretty sure you will do this much more effectively than me.

File: sphinx/source/docs/user_guide/examples/extensions_putting_together_ts.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    this.render()
+
+    // Set Backbone listener so that when the Bokeh slider has a change
+    // event, we can process the new data
+    this.connect(this.model.slider.change, () => this.render())
+  }
+
+  render() {
+    // Backbone Views create <div> elements by default, accessible as @el.
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/examples/extensions_putting_together_ts.py]: No more backbone

**Comment by bryevdv** [comment]
Sure I will take a look this week

**Comment by bryevdv** [comment]
@mattpap I actually replaced the coffee script version entirely. I think we should encourage users to use TypeScript as much as possible. We should update the other examples in the extensions gallery as time permits.

**Comment by bryevdv** [comment]
(close was accidental)

**Comment by mattpap** [comment]
>  I actually replaced the coffee script version entirely. I think we should encourage users to use TypeScript as much as possible.

@bryevdv, works for me.

**Pull request merged**